enumerable_methods.rb

```
@@ -0,0 +1,126 @@
+    i = 0
+    while i < array.length
+      yield array[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    array = to_a.dup
+    indexer = 0
+    i = 0
+    while i < array.length
+      yield(array[i], indexer)
+      indexer += 1
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    array = []
+    my_each { |i| array << i if yield(i) }
+    array
+  end
+
+  def my_all?(param = nil)
+    if block_given?
+      my_each { |i| return false unless yield(i) }
+    elsif param.nil?
+      my_each { |i| return false unless i }
+    else
+      my_each { |i| return false unless check_pattern(i, param) }
+    end
+    true
+  end
+
+  def my_any?(param = nil, &block)
+    if block
+      my_each { |i| return true if block.yield(i) }
+    elsif param.nil?
+      my_each { |i| return true if i }
+    else
+      my_each { |i| return true if check_pattern(i, param) }
+    end
+    false
+  end
+
+  def my_none?(param = nil, &block)
+    !my_any?(param, &block)
+  end
+
+  def my_count(param = nil)
+    indexer = 0
+    if block_given?
+      my_each { |i| indexer += 1 if yield(i) == true }
+    elsif param.nil?
+      my_each { indexer += 1 }
+    else
+      my_each { |i| indexer += 1 if i == param }
+    end
+    indexer
+  end
+
+  def my_map
+    return to_enum(:my_map) unless block_given?
+
+    new_array = []
+    my_each do |value|
+      new_array << if is_a? Hash yield(value[0], value[1])
+                   else
+                     yield(value)
+                   end
+    end
+    new_array
+  end
+
+  def my_inject(param_1 = nil, param_2 = nil, &block)
+    array = to_a.dup
+    (inject, var, array) = get_inject_and_sym(param_1, param_2, array, block_given?)
+    inject = if var inject_var(array, var, inject)
+             else inject_block(array, inject, &block)
+             end
+    inject
+  end
+end
```

- [ ] The method doesn't allow a proc or block given as arguments. Kindly check that.

- [ ] When no block or arguments are given it should raise an error. Kindly check that.